infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -163,6 +133,8 @@ public void reset(long topologyId, long snapshot) {
         phase = Phase.TRANSFER_PHASE;
         snapshotSyncStartMarker = Optional.empty();
         replicatedStreamIds.clear();
+        // Signal the config to sync with registry table to avoid data loss
+        config.syncWithRegistry();
```

Comment:
reset() is invoked at the start of the transfer phase, which can potentially take longer to complete, increasing the chance of having stale entries for the streams to replicate.

Additionally, it is also invoked when snapshot sync apply is resumed on a restart.

The 2 stages of syncing with the registry table are inconsistent.

Reply:
I think there should be a sync after the transfer phase completes because we can potentially be in this phase for a long time.

Reply:
I'm actually not convinced on how this prevents data loss like suggested in the comment. From what I see this only prevents taking in a stream during transfer phase that locally is no longer federated (is_federated = false) during upgrade. Anyways it won't be applied when shadow stream is applied to regular streams (flip). 

I'm not suggesting to remove, just that the reason I'm not sure if it's the righ one (the oe stated in the comment).